File: CSharp/WebSockets/ClientEnterpriseEdition.cs
using System;

namespace ThingModel.WebSockets
{
    public class ClientEnterpriseEdition : Client
    {
        private bool _isLive = true;
        private bool _isPaused = false;

        public bool IsLive
        {
            get { return _isLive; }
            private set { _isLive = value; }
        }

        public bool IsPaused
        {
            get { return _isPaused; }
            private set { _isPaused = value; }
        }

        public ClientEnterpriseEdition(string senderID, string path, Warehouse warehouse) : base(senderID, path, warehouse)
        {
        }


        public void Live()
        {
            if (IsLive && !IsPaused) return;

            Send("live");
            IsLive = true;
            IsPaused = false;
        }

        public void Play()
        {
            if (!IsPaused) return;

            Send("play");
            IsPaused = false;
        }

        public void Pause()
        {
            if (IsPaused) return;

            Send("pause");
            IsPaused = true;
        }

		private static readonly TimeSpan DateTimeEpoch = new TimeSpan(
			new DateTime(1970,1,1,0,0,0, DateTimeKind.Utc).Ticks);

        public void Load(DateTime time)
        {
            IsLive = false;
            IsPaused = true;

	        Send("load "+time.Subtract(DateTimeEpoch).Ticks/10000);
        }

        protected override void WsOnStringMessage(string message)
        {
            if (message.Contains("error"))
            {
                Console.WriteLine("ThingModelClientEnterpriseEdition "+message);
            }
        }

        public override void Send()
        {
            if (!IsLive || IsPaused)
            {
                throw new Exception("ThingModelClientEnterpriseEdition cannot send data while paused or in a past situation");
            }

            base.Send();
        }
    }
}

File: TypeScript/WebSockets/ClientEnterpriseEdition.ts
 module ThingModel.WebSockets {
	 export class ClientEnterpriseEdition extends Client {
		 private _isLive: boolean;

		 public get IsLive(): boolean {
			 return this._isLive;
		 }

		 private _isPaused: boolean;

		 public get IsPaused(): boolean {
			 return this._isPaused;
		 }

		 constructor(senderID: string, path: string, warehouse: Warehouse) {
			 super(senderID, path, warehouse);

			 this._isLive = true;
			 this._isPaused = false;
		 }

		 public Live(): void {
			 if (this._isLive && !this._isPaused) return;

			 this.SendMessage("live");

			 this._isLive = true;
			 this._isPaused = false;
		 }

		 public Play(): void {
			 if (!this._isPaused) return;

			 this.SendMessage("play");

			 this._isPaused = false;
		 }


		 public Pause(): void {
			 if (this._isPaused) return;

			 this.SendMessage("pause");

			 this._isPaused = true;
		 }

		 public Load(time: Date): void {
			 this.SendMessage("load " + time.getTime());

			 this._isLive = false;
			 this._isPaused = true;
		 }

		 public Send(): void {
			 if (!this._isLive || this._isPaused) {
				 console.debug("ThingModelClientEnterpriseEdition cannot send data while paused or in a past situation");
				 return;
			 }
			 super.Send();
		 }
	 }
 }
